### PR TITLE
Add docs for telemetry configuration

### PR DESCRIPTION
**Pull request by ajgajg1134**
Adds DD_INSTRUMENTATION_TELEMETRY_ENABLED variable details.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/library_config/go.md [hunk 1]
```diff
@@ -137,3 +140,4 @@ extracted value is used.
 [4]: /tracing/trace_pipeline/ingestion_mechanisms/?tab=go#pagetitle
 [5]: /tracing/advanced/setting_primary_tags_to_scope/#environment
 [6]: /getting_started/tracing/#environment-name
+[7]: /tracing/configure_data_security/?tab=mongodb#telemetry-collection
```
Comment (katiehockman) [inline content/en/tracing/trace_collection/library_config/go.md:143]:
```suggestion
[7]: /tracing/configure_data_security#telemetry-collection
```

File: content/en/tracing/trace_collection/library_config/go.md [hunk 2]
```diff
@@ -102,6 +102,9 @@ Enable web framework and library instrumentation. When false, the application co
 : **Default**: `null` <br>
 Dynamically rename services through configuration. Services can be separated by commas or spaces, for example: `mysql:mysql-service-name,postgres:postgres-service-name`, `mysql:mysql-service-name postgres:postgres-service-name`.
 
+`DD_INSTRUMENTATION_TELEMETRY_ENABLED`
+: **Default**: `false` <br>
+Datadog may collect [some telemetry][7]. This variable may be changed to default `true` in the future. To disable this telemetry collection, explicitly set this variable to `false`.
```
Comment (katiehockman) [inline content/en/tracing/trace_collection/library_config/go.md:107]:
```suggestion
Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. 
When false, this telemetry data will not be collected.
```

Comment (katiehockman) [reply content/en/tracing/trace_collection/library_config/go.md:107]: If/when it is turned on by default in the future, that should go in release notes, and we should inform all customers in advance. We can update the docs at that time as well.

File: content/en/tracing/trace_collection/library_config/go.md [hunk 3]
```diff
@@ -102,6 +102,10 @@ Enable web framework and library instrumentation. When false, the application co
 : **Default**: `null` <br>
 Dynamically rename services through configuration. Services can be separated by commas or spaces, for example: `mysql:mysql-service-name,postgres:postgres-service-name`, `mysql:mysql-service-name postgres:postgres-service-name`.
 
+`DD_INSTRUMENTATION_TELEMETRY_ENABLED`
+: **Default**: `false` <br>
+Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. 
+When false, this telemetry data will not be collected.
```
Comment (jtappa) [inline content/en/tracing/trace_collection/library_config/go.md:108]:
```suggestion
Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. When false, this telemetry data will not be collected.
```

**Review by jtappa — APPROVED** [review]

**Review by katiehockman — APPROVED** [review]

**Pull request merged**